src/__checks__/select.check.ts

```diff
@@ -42,201 +42,212 @@ const bar = defineTable({
 });
 
 test('should output all columns and the data type', () => {
-  expectType<{
+  expect(toTableRow(foo)).type.toEqual<{
     id: string;
     createDate: Date;
     name: string;
     value: number | null;
-  }>(toTableRow(foo));
+  }>();
 });
 
 const db = defineDb({ foo, bar }, () => Promise.resolve({ rows: [], affectedCount: 0 }));
 
 describe('select', () => {
   test('should return null and not null properties', () => {
-    expectType<{
+    expect(
+      toSnap(db.select(db.foo.id, db.foo.createDate, db.foo.value).from(db.foo)),
+    ).type.toEqual<{
       id: string;
       createDate: Date;
       value: number | null;
-    }>(toSnap(db.select(db.foo.id, db.foo.createDate, db.foo.value).from(db.foo)));
+    }>();
   });
 
   test('should return nullable properties of left joined columns', () => {
-    expectType<{
+    expect(
+      toSnap(db.select(db.foo.id, db.bar.endDate, db.bar.value).from(db.foo).leftJoin(db.bar)),
+    ).type.toEqual<{
       id: string;
       endDate: Date | null;
       value: number | null;
-    }>(toSnap(db.select(db.foo.id, db.bar.endDate, db.bar.value).from(db.foo).leftJoin(db.bar)));
+    }>();
   });
 
   test('should return nullable properties of left side properties when right joining', () => {
-    expectType<{
+    expect(
+      toSnap(db.select(db.foo.name, db.bar.startDate, db.bar.value).from(db.foo).rightJoin(db.bar)),
+    ).type.toEqual<{
       value: number | null;
       name: string | null;
       startDate: Date;
-    }>(
-      toSnap(db.select(db.foo.name, db.bar.startDate, db.bar.value).from(db.foo).rightJoin(db.bar)),
-    );
+    }>();
   });
 
   test('should select * and return nullable properties of left side properties when right joining', () => {
-    expectType<{
-      id: string | null;
+    expect(toSnap(db.select(star()).from(db.foo).rightJoin(db.bar))).type.toEqual<{
+      id: unknown;
```

Comment:
These changes to `unknown` are caught in the newer version of typescript.

The problem is with queries that use `star` with a join, and the resulting columns have overlapping names, so the resulting type can be ambiguous.  In the existing examples, both tables have the same types for these overlapping columns, so this previously used to work.  Now, it appears that typescript is catching this and returning an `unknown` type instead.

When a query is run with overlapping column names, the order of the columns in the select determines which value will be used.  It is (maybe) possible to update the types to take order into account, and overwrite types based on the order, but this seems like a pretty large change to make.

I think that leaving these overlapping names returning `unknown` instead is safer, and can avoid errors from overlapping by mistake.